modules/apps/bean-portlet/bean-portlet-registration-impl/src/main/java/com/liferay/bean/portlet/registration/internal/BeanPortletRegistrarImpl.java

```diff
@@ -320,19 +152,52 @@ public void step4ApplicationScopedInitializedAsync(
 			}
 		}
 
-		Function<String, Set<BeanMethod>> portletBeanMethodsFunction =
-			_collectPortletBeanMethods(beanManager);
+		List<DiscoveredBeanMethod> discoveredBeanMethods = new ArrayList<>();
+
+		for (Class<?> discoveredClass : discoveredClasses) {
+			try {
+				for (Method method : discoveredClass.getMethods()) {
+					for (BeanPortletMethodType beanPortletMethodType :
+							BeanPortletMethodType.values()) {
+
+						if (beanPortletMethodType.isMatch(method)) {
+							discoveredBeanMethods.add(
+								new DiscoveredBeanMethod(
+									discoveredClass, method,
+									beanPortletMethodType));
+						}
+					}
+				}
+			}
+			catch (Throwable t) {
```

Comment:
@ngriffin7a what Throwable we are expecting here? The only code can throw something is "discoveredClass.getMethods()" when this is a security manager stopping it. I doubt the need of this catch block.

Reply:
@shuyangzhou I don't recall the specific Throwable that we were expecting. But I do recall that the Spring Framework was throwing exceptions that we are able to ignore -- that's really the point of the catch block -- to ignore the exception thrown by Spring. I don't think Spring was throwing the exception because of a SecurityManager. If you want me to debug this and get more specific I can. Just let me know.

Reply:
Can you get me steps to trigger the Throwables?

Reply:
@shuyangzhou Yes, happy to do that. But it will be a few hours because I have to perform some JSF testing locally for Clarissa because there is a problem running the JSF tests in CI and she's trying to get a release out the door.

Reply:
No hurry, I can focus on other parts first.

Reply:
@shuyangzhou Actually, I think that simply deploying the following portlet might trigger it: https://github.com/ngriffin7a/liferay-blade-samples/tree/applicant/maven/apps/applicant/applicant-mvcbean-spring-jsp-portlet

If not, then try the Thymeleaf version:
https://github.com/ngriffin7a/liferay-blade-samples/tree/applicant/maven/apps/applicant/applicant-mvcbean-spring-thymeleaf-portlet

Reply:
I deployed all 4 of the demo wars, nothing of them run into this. Please double check, if you don't see it either, then this should be removed.

Reply:
@shuyangzhou Did you try putting the Spring ones on a portal page and viewing the page in the browser? I can't remember, but it might be necessary to hit the page for the first request.

Reply:
yes, I did. I have them on the page, nothing hits that catch block.

Reply:
@shuyangzhou OK that's all I would have done to try and reproduce it. Perhaps it was a problem when I originally targeted Spring Framework 4.x but this PR targets 5.x. So I guess it's fine to remove the try/catch. Thanks.

Reply:
ok, I will remove it then.